src/state.ts

```diff
@@ -21,134 +21,181 @@ export class StateManager {
         return this.extensionContext.globalState.update(this.folder.uri.fsPath + key, value);
     }
 
+    setIsMultiProject(isMultiProject: boolean) {
+        this.isMultiProject = isMultiProject;
+    }
+
     /**
      * Whether the user chose to ignore the popup message about missing CMakeLists.txt
      * from the root folder, for a code base that is not fully activating CMake Tools.
      */
-    get ignoreCMakeListsMissing(): boolean {
-        return this._get<boolean>('ignoreCMakeListsMissing') || false;
+    getIgnoreCMakeListsMissing(folderName: string): boolean {
+        return this.isMultiProject ? (this._get<boolean>(`${folderName} ignoreCMakeListsMissing`) || false) :
+            (this._get<boolean>('ignoreCMakeListsMissing') || false);
     }
 
-    async setIgnoreCMakeListsMissing(v: boolean) {
-        await this._update('ignoreCMakeListsMissing', v);
+    async setIgnoreCMakeListsMissing(folderName: string, v: boolean) {
```

Comment:
`folderName` cannot be trusted to be unique across multiple projects. It would be better to pass in some other information here instead. You could also consider passing in the `CMakeProject` instance and pick the data you want here instead of extracting the information first. It signals to the caller that the information should be coming from the CMake Project object (though I'm not sure if all the callers have access to that or not).

Reply:
`setIgnoreCMakeListsMissing` is only called in `reset` which is only called on CMakeProject. I think it is reliable in this case.
In this new design, we are saving a different value for `ignoreCMakeListsMissing` for every project. previously we were only saving one variable for all projects.

Reply:
marking resolved due to `ignoreCmakeListsMissing` being unique for each project